

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Raw reads &mdash; Bioinformatics | RBGE  documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" type="text/css" />
  <link rel="stylesheet" href="_static/panels-bootstrap.5fd3999ee7762ccc51105388f4a9d115.css" type="text/css" />
  <link rel="stylesheet" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="FastQC" href="fastqc.html" />
    <link rel="prev" title="Installing Software" href="installsoftware.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> Bioinformatics | RBGE
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">Wet Laboratory:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="wet_labbook_template.html">Wet Lab Notebook Template</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Data management:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="wheretostore.html">Where do I store my data?</a></li>
<li class="toctree-l1"><a class="reference internal" href="wheretorun.html">Where do I run my analysis?</a></li>
<li class="toctree-l1"><a class="reference internal" href="installsoftware.html">Installing Software</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Quality of the raw reads:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Raw reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="fastqc.html">FastQC</a></li>
<li class="toctree-l1"><a class="reference internal" href="trimmomatic.html">Trimmomatic</a></li>
<li class="toctree-l1"><a class="reference internal" href="cutadapt.html">Cutadapt</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Target Capture:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Mapping reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="aligning.html">Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogeny.html">Inferring a phylogeny</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Contact:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="contact.html">Getting in touch</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Bioinformatics | RBGE</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Raw reads</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/rawreads.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="raw-reads">
<h1>Raw reads<a class="headerlink" href="#raw-reads" title="Permalink to this headline">¶</a></h1>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Never work directly on your raw data. Make a copy of it to a working directory (i.e. where you will run your analysis) and keep backup copies of your raw data elsewhere (see <a class="reference internal" href="wheretostore.html"><span class="doc">Where do I store my data?</span></a>).</p>
</div>
<p>For this tutorial, we will follow the Hyb_baits_pipeline from the paper <a class="reference external" href="https://www.frontiersin.org/articles/10.3389/fpls.2015.00710/full">Nicholls et al. 2015</a>, with scripts available on github: <a class="reference external" href="https://github.com/ckidner/Targeted_enrichment">https://github.com/ckidner/Targeted_enrichment</a>. We will use five accessions used in the manuscript, available in European Nucleotide Archive, study ID ERP009747: <a class="reference external" href="https://www.ebi.ac.uk/ena/browser/view/PRJEB8722?show=reads">https://www.ebi.ac.uk/ena/browser/view/PRJEB8722?show=reads</a>.</p>
<p>Here we have a list of five folders that came from the sequencing facility. The name of the folders represent the name of the accessions we are working with. Inside each folder are the forward and reverse reads of the accessions. Let’s see what their names are:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ls FG35/
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>140428_M01145_0124_000000000-A78G3_1_IL-TP-005_1.sanfastq.gz
140428_M01145_0124_000000000-A78G3_1_IL-TP-005_2.sanfastq.gz
</pre></div>
</div>
<p>The files have complicated names. Let’s change their names to more meaningful ones (i.e., the name of the accessions, which is the same name of their folders):</p>
<p>Let’s start by making a list of folders we have in our working directory:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>find . -type d
</pre></div>
</div>
<p>This command is saying find here (the dot <code class="docutils literal notranslate"><span class="pre">.</span></code>) all the directories <code class="docutils literal notranslate"><span class="pre">-type</span> <span class="pre">d</span></code>. Here is what you will see in your terminal:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./FGIntype
./KGD465
./FG35
./zygia917
./FG113
</pre></div>
</div>
<p>Let’s delete the symbols that we don’t need ( <cite>.</cite> and <cite>/</cite> ). We will use a pipe <code class="docutils literal notranslate"><span class="pre">|</span></code> and the command <code class="docutils literal notranslate"><span class="pre">cut</span></code>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>find . -type d <span class="p">|</span> cut -c <span class="m">3</span>-
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>FG113
FG35
FGIntype
KGD465
zygia917
</pre></div>
</div>
<p>Now let’s save their names in a file. The symbol <code class="docutils literal notranslate"><span class="pre">&gt;</span></code> will redirect our output to a file called <code class="docutils literal notranslate"><span class="pre">acc</span></code>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>find . -type d <span class="p">|</span> cut -c <span class="m">3</span>- &gt; acc
</pre></div>
</div>
<p>Take a look at <code class="docutils literal notranslate"><span class="pre">acc</span></code>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cat acc
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>FG113
FG35
FGIntype
KGD465
zygia917
</pre></div>
</div>
<p>We will now use the script <code class="docutils literal notranslate"><span class="pre">renaming.sh</span></code> to rename our raw read and move them to the folder we are working on.</p>
<details class="sphinx-bs dropdown card mb-3 shadow">
<summary class="summary-title card-header text-info">
You can see the script in this dropdown menu. The file is also in the GitHub repository:<div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<blockquote>
<div><div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash -x</span>
<span class="c1"># Renaming the files by the folder names, getting them out of the folders</span>
<span class="c1"># For dealing with sequencing facility output</span>

<span class="c1"># In directory with the folders in make a list of folder names, to run through this on a while loop</span>

<span class="nv">acc</span><span class="o">=</span><span class="nv">$1</span>

<span class="nv">input1</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>/*_1.sanfastq.gz
<span class="nv">output1</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>_1.fastq.gz

<span class="nv">input2</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>/*_2.sanfastq.gz
<span class="nv">output2</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>_2.fastq.gz

mv <span class="nv">$input1</span> <span class="nv">$output1</span>
mv <span class="nv">$input2</span> <span class="nv">$output2</span>
</pre></div>
</div>
</div></blockquote>
</div>
</details><div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>./renaming.sh FG35
</pre></div>
</div>
<p>If you have many samples, this task can take a while. Let’s use the script <code class="docutils literal notranslate"><span class="pre">renaming.sh</span></code> but in a loop in order to change the names of all the accessions with a single command. Loops are nice to automate repetitive tasks:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="k">while</span> <span class="nb">read</span> f<span class="p">;</span> <span class="k">do</span> ./renaming.sh <span class="s2">&quot;</span><span class="nv">$f</span><span class="s2">&quot;</span><span class="p">;</span> <span class="k">done</span> &lt; acc
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>In this loop, the <code class="docutils literal notranslate"><span class="pre">acc</span></code> is the input file that will be read one line at a time, and that value replaces the variable <code class="docutils literal notranslate"><span class="pre">$f</span></code>. So if our <code class="docutils literal notranslate"><span class="pre">acc</span></code> files has five lines, the command <code class="docutils literal notranslate"><span class="pre">./renaming.sh</span></code> will be executed five times, each time replacing <code class="docutils literal notranslate"><span class="pre">$f</span></code> with a line from our input file <code class="docutils literal notranslate"><span class="pre">acc</span></code>.</p>
</div>
<details class="sphinx-bs dropdown card mb-3">
<summary class="summary-title card-header">
This is what you should see printed in your terminal:<div class="summary-down docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="summary-up docutils">
<svg version="1.1" width="24" height="24" class="octicon octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="summary-content card-body docutils">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>+ <span class="nv">acc</span><span class="o">=</span>FG113
+ <span class="nv">input1</span><span class="o">=</span><span class="s1">&#39;FG113/*_1.sanfastq.gz&#39;</span>
+ <span class="nv">output1</span><span class="o">=</span>FG113_1.fastq.gz
+ <span class="nv">input2</span><span class="o">=</span><span class="s1">&#39;FG113/*_2.sanfastq.gz&#39;</span>
+ <span class="nv">output2</span><span class="o">=</span>FG113_2.fastq.gz
+ mv FG113/140428_M01145_0124_000000000-A78G3_1_IL-TP-019_1.sanfastq.gz FG113_1.fastq.gz
+ mv FG113/140428_M01145_0124_000000000-A78G3_1_IL-TP-019_2.sanfastq.gz FG113_2.fastq.gz
+ <span class="nv">acc</span><span class="o">=</span>FG35
+ <span class="nv">input1</span><span class="o">=</span><span class="s1">&#39;FG35/*_1.sanfastq.gz&#39;</span>
+ <span class="nv">output1</span><span class="o">=</span>FG35_1.fastq.gz
+ <span class="nv">input2</span><span class="o">=</span><span class="s1">&#39;FG35/*_2.sanfastq.gz&#39;</span>
+ <span class="nv">output2</span><span class="o">=</span>FG35_2.fastq.gz
+ mv FG35/140428_M01145_0124_000000000-A78G3_1_IL-TP-005_1.sanfastq.gz FG35_1.fastq.gz
+ mv FG35/140428_M01145_0124_000000000-A78G3_1_IL-TP-005_2.sanfastq.gz FG35_2.fastq.gz
+ <span class="nv">acc</span><span class="o">=</span>FGIntype
+ <span class="nv">input1</span><span class="o">=</span><span class="s1">&#39;FGIntype/*_1.sanfastq.gz&#39;</span>
+ <span class="nv">output1</span><span class="o">=</span>FGIntype_1.fastq.gz
+ <span class="nv">input2</span><span class="o">=</span><span class="s1">&#39;FGIntype/*_2.sanfastq.gz&#39;</span>
+ <span class="nv">output2</span><span class="o">=</span>FGIntype_2.fastq.gz
+ mv FGIntype/140428_M01145_0124_000000000-A78G3_1_IL-TP-009_1.sanfastq.gz FGIntype_1.fastq.gz
+ mv FGIntype/140428_M01145_0124_000000000-A78G3_1_IL-TP-009_2.sanfastq.gz FGIntype_2.fastq.gz
+ <span class="nv">acc</span><span class="o">=</span>KGD465
+ <span class="nv">input1</span><span class="o">=</span><span class="s1">&#39;KGD465/*_1.sanfastq.gz&#39;</span>
+ <span class="nv">output1</span><span class="o">=</span>KGD465_1.fastq.gz
+ <span class="nv">input2</span><span class="o">=</span><span class="s1">&#39;KGD465/*_2.sanfastq.gz&#39;</span>
+ <span class="nv">output2</span><span class="o">=</span>KGD465_2.fastq.gz
+ mv KGD465/140428_M01145_0124_000000000-A78G3_1_IL-TP-020_1.sanfastq.gz KGD465_1.fastq.gz
+ mv KGD465/140428_M01145_0124_000000000-A78G3_1_IL-TP-020_2.sanfastq.gz KGD465_2.fastq.gz
+ <span class="nv">acc</span><span class="o">=</span>zygia917
+ <span class="nv">input1</span><span class="o">=</span><span class="s1">&#39;zygia917/*_1.sanfastq.gz&#39;</span>
+ <span class="nv">output1</span><span class="o">=</span>zygia917_1.fastq.gz
+ <span class="nv">input2</span><span class="o">=</span><span class="s1">&#39;zygia917/*_2.sanfastq.gz&#39;</span>
+ <span class="nv">output2</span><span class="o">=</span>zygia917_2.fastq.gz
+ mv zygia917/140428_M01145_0124_000000000-A78G3_1_IL-TP-027_1.sanfastq.gz zygia917_1.fastq.gz
+ mv zygia917/140428_M01145_0124_000000000-A78G3_1_IL-TP-027_2.sanfastq.gz zygia917_2.fastq.gz
</pre></div>
</div>
</div>
</details><p>Now let’s see what we have in our working directory:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ls
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>FG113               FG35_2.fastq.gz     KGD465_1.fastq.gz   zygia917
FG113_1.fastq.gz    FGIntype            KGD465_2.fastq.gz   zygia917_1.fastq.gz
FG113_2.fastq.gz    FGIntype_1.fastq.gz acc                 zygia917_2.fastq.gz
FG35                FGIntype_2.fastq.gz FG35_1.fastq.gz     KGD465              renaming.sh
</pre></div>
</div>
<p>The raw reads are in <code class="docutils literal notranslate"><span class="pre">fastq.gz</span></code> format. <code class="docutils literal notranslate"><span class="pre">fastq</span></code> is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores. The <code class="docutils literal notranslate"><span class="pre">gz</span></code> means the files are compressed. We can peek into compressed files with <code class="docutils literal notranslate"><span class="pre">zless</span></code>. Let’s look into the forward reads (forward reads end with <code class="docutils literal notranslate"><span class="pre">_1</span></code> in our case) of the accession FG113.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>zless FG113_1.fastq.gz
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>@HWI-M01145:124:000000000-A78G3:1:1101:15884:1359 <span class="m">1</span>:N:0:GTGAAA
AATCTAAAACCACCTCAATAAATGTAATCTTCATCTATCTTATCTATTTAAAGTCCTTATTATTCCTTTTGTTTCTTCAAATCCAGCTCCCTCTTCTCTCCTTTCTCAAGTATCAATATGTTGAGATGCATCAATACTTCTTCAGTTCGTTCTCCTTTTCCTCTTCCGCCTCTACAGCCTCTACCGCATCTACCTCCTCTACTGCAACCGTCAGAGTGACAAGTTCTTGACCACCTCAATGCGAGGGTGG
+
BBBBBFFFFFABGGGGGGGGGGGHHFAFAFGFGHHHHHHHHHGHHHHHHHHHHGGHHHHHHHHHHHHHHHHHHHHHHHGHHHHGHHGHHHGHHHHHGHHHHHHHHHHGHHFHHHHHHHHHHHGFHHGHHGHGHHHHHGHHHHHHHHHHGGHHGHHHHHHHHHHHHHHGGGGGHHHHGHHHHHHHHGGGGGHHHHHHHHHHHHHHHHHHHGGGGGGHFHHHGHHHHHHHHHHHHHGHHHHHHHGGGGGE?D
@HWI-M01145:124:000000000-A78G3:1:1101:14180:1422 <span class="m">1</span>:N:0:GTGAAA
TTGAGATCAACTTGGTCATCCTTGTCGCGTCCCAATATCATCTTATAAACAATATATCTTCTAAAAGCCCATGCATTTTCTCACTAACGTCCACAATCTCACCCATCTTAGCAGCATTATCCAGTGACTTCACCACCATTTTCAACTCCTCCGATCCCTCCTAAGATGACCAAACGGCTCTTGATTTGAAGCACCAAAAAGATGTTGGAGGCTGAGTTTCTTCATGTCACGCAAGTATGAACCATGCCCA
+
&gt;AAAAAFFFF1DEGGFGFFFGG311FEEAFGEA1FGGHHDEGGEAEHF2GDCFHHHHHHFHHBAHFGFFEHHBF1AGHHH12GHHHHFCFEFEFHHFGHHGHFGGGHHHHGHBGEGHHHHHGHH2GFFFGGFHGGHGHHHHHHHHGHHFFHHFCAC/0FFECGHH1GD1FGFH/G&lt;EC?CHHD&lt;GGHFBFGFGHEFC..&lt;A.D&lt;GDCE0C?.GFHHHFHHHHBGHFF09FB?@AG0FFGFFFAABFFGGE
</pre></div>
</div>
<p>To exit <code class="docutils literal notranslate"><span class="pre">zless</span></code> just type <code class="docutils literal notranslate"><span class="pre">q</span></code>.</p>
<p>Each read has four lines of information, so here we see two reads. In the first line beginning with <code class="docutils literal notranslate"><span class="pre">&#64;</span></code> we see information from the sequencing machine (in this case information about the flowcell in a Illumina machine), the nucleotide sequence is stored on the second line, the third line has a <code class="docutils literal notranslate"><span class="pre">+</span></code> and the fourth line stores the quality scores.</p>
<p>We are ready to the next step. Let’s run <a class="reference internal" href="fastqc.html"><span class="doc">FastQC</span></a> on them.</p>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="fastqc.html" class="btn btn-neutral float-right" title="FastQC" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="installsoftware.html" class="btn btn-neutral float-left" title="Installing Software" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021 | Bioinformatics | Royal Botanic Garden Edinburgh.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>