<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Mapping reads &mdash; Bioinformatics | RBGE  documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/design-style.59c74d8c95b765a7fd995ac71d459ebe.min.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/design-tabs.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Alignment" href="aligning.html" />
    <link rel="prev" title="Cutadapt" href="cutadapt.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> Bioinformatics | RBGE
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Wet Laboratory:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="wet_labbook_template.html">Wet Lab Notebook Template</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Data management:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="setup.html">Setting up your working environment</a></li>
<li class="toctree-l1"><a class="reference internal" href="wheretostore.html">Where do I store my data?</a></li>
<li class="toctree-l1"><a class="reference internal" href="wheretorun.html">Where do I run my analysis?</a></li>
<li class="toctree-l1"><a class="reference internal" href="installsoftware.html">Installing Software</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Quality of the raw reads:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="rawreads.html">Raw reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="fastqc.html">FastQC</a></li>
<li class="toctree-l1"><a class="reference internal" href="trimmomatic.html">Trimmomatic</a></li>
<li class="toctree-l1"><a class="reference internal" href="cutadapt.html">Cutadapt</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Target Capture:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Mapping reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="aligning.html">Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogeny.html">Inferring a phylogeny</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Contact:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="contact.html">Getting in touch</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Bioinformatics | RBGE</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Mapping reads</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/mapping.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="mapping-reads">
<h1>Mapping reads<a class="headerlink" href="#mapping-reads" title="Permalink to this headline"></a></h1>
<p>We will now map our trimmed reads to a reference. We will use as a reference the bait sequence <code class="docutils literal notranslate"><span class="pre">Ref.fna</span></code> that is inside the Targeted_enrichment repository we cloned from GitHub. Let’s take a look inside this file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>less /your/path/to/Targeted_enrichment/Ref.fna
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>&gt;comp39600_c0_seq2
CGGGCTTTAAGAGGGACCTTCTTGACAATGGAAAGCCTACAAGCCTCCTCCATATTTAAGAGGGACCTTTTTGATAAAATAAAACATATATAAATCCACTTAATCCAGTGAATACAGATGAAGAGTCAACCTAGGCTCAACCTGGGCTTCAAGAGGGACCTTCTTGGCAACGGAAAGTCCATGAGCCTCCTCCATATTCAAGTCATGCCCCTCCATGTTCCCAGCTAGCTGCCACCTGAACCCATGAACGAGATTCGCCAAACTTGATTGAATCACCCTAAGCCCGAGAGGATAGCCCGGGCACATTCTTCTTCCAGCCCCAAATGGCAACAACTCAAAGTCATGACCGTTCACATCAATCTCCTTAGTCGGGCCATCAATGAACCTCTCGGGCTTAAACTCATTGGGCCAGCCCCAAACTGTCGGGTCTCTCCCTATGGCCCACACATTCAAAATCACTTGGGTGCCTTTGGGAACGTCATAGCCATCAATTTTACAGTGCTCCCGGGCTTCCCTTGGGATCAGTGTGGGTGCCACAGGGTGTAACCTCATTGTTTCTTTAACAATTGCTTTAATGTAAGGTAGATTGACCATGTCTTGTTCTTCAACCCATCTGTTTCTACCTACCACATTGTCTAGCTCTTGTTGTGCCTGCTTCAAGACTTGTGGCTTTTTAAGAAGCTCTGACATTGCCCATTCCACTGTTATAGAAGAGCTCTGCGTCCCCCCTATGAGAAGATCCTGAGTAATTGCTTTCACATTATGCTTTTGAAGCTTGACCTCGTGGCTAGGATCATCTGCAGCTTGCAAAAGGAGATCCACCATATCCCTGGCAACATAATTCTTAATCCCTCTTCTTCTGTCATTGTGTTCATTCAATATATGTTCCAAAAACCTCTCAAATTTCTTACTCACGACCTTCATCCTCTTCACATATCCCTGCAAATCCAACAATTCAAGCCAAGGAATGAAATCCCCAATATTCAGAAACCCACTATTCAATAACAGAAACTCATCTAGCATCCACTTGAACTCATCCGGCGTAATAACAGCATTTTCGGTTTTGTTAACGTATGTCTTCCCCAAAACCATGCGACTTATGATGTTGAGGTTGAAAAGGTTGAGGTAGTCCTTTAAGAATATGGTTTTGCCCGAAGAACTGTACAGTTGATTTGCCATGATTCTGATCTCTTCCTTTCTGATATACTCATAGGACTCCAAGCTTTTGGGAGAGAAGAGCTCGGTAAAACAGAGTTTGCGGGCTTGGCGCCAGTATGGGGAGTATGGGGCCCAGATAATGTTGGAGTGGTCGAAACTTGTGTACTTACCGGCCGCAGTTCTCGGCCGCGACGCTAAGGAGCCGTCGTGGGTTTGCAGAAAGGCCTTAGCCATTTTCGCTGAGGAGCCAACGACGACAGACTTTGAGCCTAACCAGAGGTGCATGACGGGTCCGTACTTTTGGGATAGAACATGGAGGGACTTGTGGGGGAATAGGCCTATGAGGTTGAGGTTTCCGA
&gt;comp39985_c0_seq4
TTTTTTTATTGTAAAGGGGAGATTTAGTATTGCATTATTTCTTCCATTGTCAAAATGCAAAGGATTACAAGGATCTACAGCTTCGTAGCTGCTTCCATGAAATGTTTCAGTGTTTGGCACTGAAAACAATTCGAATATTTATTTAAAACAAACCAAACTCATTAAAGGCATACATATGTTGGTTAAAAAAAAGAAAAAAAAAAGGCATACATAAAATACACCTCTAAAGAATGCCAATGTTTGATCTAGCACCCTGCATAGTGGCCATCTTCACCCCTCAAAAAGCACTCAATTCGTTCTAGCGCCTCAATAGCAATGGTAGTCGGTGAGATATCTGCAACATCTCTACAGCCCAGTTTTGCAGCAATATTTTCTAAGGACACTCGAGCATTGGCATTGGCATAAGCTTCGCCTCTTTCTTCGAAAAGTGTAGACAAACGCATCAAAGCCCGCTTGTACGCATCTCCTGCTTCATAATGCAGAAGTGGACGAGAATTAGTTCCCACAGCTGCAATTCTCCTGGCCAGGGCTTCTAAAGGTACATCCAACCAAACAGTGATCCCCTTGTGCATATATTTCCAATTGATGGGCCTCACAACTGCACCTCCACCAGTAGAAATTACAAGTCCATGCATAAGTGACAGATTACGCAATACCTCAGTCTCCTTGTCCCTGAAGAAATCCTCTCCATAGTGCTTGAATATATCAGCTACAGAAGTTCCACCAACCTCATGCTCCACCAAGGCATCACTATCAAAAAAGGAATAACCAAGCACTTGTGACAAAATCTTCCCCACAGTTGTCTTCCCTGACCCCATCATTCCAACAAGATATATACAGCGTCCATTTAAATAAGGCTCGATCTCTTGTGTCCTAGTCTGTAACATTAACTCTTCATCAGGAGGGGCCTCTAAGCTTCCAGAGTCCAATGTCAAAGCTGGCAAATTGTCGTAGGAACAAGAAACCTCCAAAGGGGCCGCCCGCCCCCGATTGCGGGAAGGTCTTGCAGTTGCAGGCTGCAAAACACAAATCCGAAGCTTCTGCGGCTCTTCAAAAACTCTAGAGAATTTCACAGAACCACTTAATTTCCTCCCAATCTTCTCGTAATGAATCAATGTTGAGAATTGCAGCCTCTGAGCCGCTTTTGCATCCATGGCGTCGCCGCAAAATAACTGACAACCAGCGTGATATAGCAGTGAACGCAGAGTTTTCAGCACATATACTCAATCCAGATTGACGAGCAGAAAGAAATAAAACGAGAAAATTAATGAACGAAACAGCTCCAGAAATAGGCACGTAAAAG
</pre></div>
</div>
<p>This is the sequence of our baits. It has the bait name (e.g. <code class="docutils literal notranslate"><span class="pre">&gt;comp39600_c0_seq2</span></code> followed by the sequence itself). We can count how many baits we have in this file by counting the number of <code class="docutils literal notranslate"><span class="pre">&gt;</span></code> since each bait name starts with this symbol:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cat /your/path/to/Targeted_enrichment/Ref.fna <span class="p">|</span> grep -c <span class="s2">&quot;&gt;&quot;</span>
</pre></div>
</div>
<p>So we have 264 <em>loci</em> in our bait set. To map our trimmed reads to the bait set we will use the script <code class="docutils literal notranslate"><span class="pre">bam_me.sh</span></code>.</p>
<details class="sd-sphinx-override sd-dropdown sd-card sd-mb-3">
<summary class="sd-summary-title sd-card-header sd-bg-info sd-bg-text-info">
bam_me.sh<div class="sd-summary-down docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="sd-summary-up docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="sd-summary-content sd-card-body docutils">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash -x</span>
<span class="c1"># Catherine Kidner 28 Oct 2014</span>
<span class="c1"># Assumes all your reads are in a folder called trimmed and all are gzipped.</span>

<span class="nb">echo</span> <span class="s2">&quot;Hello world&quot;</span>

<span class="nv">acc</span><span class="o">=</span><span class="nv">$1</span>

<span class="nb">echo</span> <span class="s2">&quot;Working on </span><span class="nv">$1</span><span class="s2">&quot;</span>
<span class="nv">score</span><span class="o">=</span>G,20,8
<span class="nv">fwd_p</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>_trimmed_1.fastq.gz
<span class="nv">rev_p</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>_trimmed_2.fastq.gz
<span class="nv">un_p</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>_trimmed_1u.fastq,<span class="si">${</span><span class="nv">acc</span><span class="si">}</span>_trimmed_2u.fastq
<span class="nv">sam</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>.sam
<span class="nv">index</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>_sorted.bam
<span class="nv">pileup</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>.pileup
<span class="nv">vcf</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>.vcf
<span class="nv">bowtie</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>_bowtie_output
<span class="nv">sorted</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>_sorted

bowtie2 --local --score-min <span class="nv">$score</span> -x ./Inga_unique_baits -1 <span class="nv">$fwd_p</span>  -2 <span class="nv">$rev_p</span>  -U <span class="nv">$un_p</span>  -S <span class="nv">$sam</span> <span class="m">2</span>&gt;<span class="nv">$bowtie</span>
samtools view -bS <span class="nv">$sam</span> <span class="p">|</span> samtools sort -o <span class="nv">$index</span> <span class="nv">$sam</span>
samtools index <span class="nv">$index</span>
samtools mpileup -E -uf Ref.fna  <span class="nv">$index</span> &gt; <span class="nv">$pileup</span>
bcftools call -c <span class="nv">$pileup</span> &gt; <span class="nv">$vcf</span>
rm *.sam
rm *.pileup

<span class="nb">exit</span> <span class="m">0</span>
</pre></div>
</div>
</div>
</details><p>We can see that this script uses two different software: Bowtie2 and Samtools.</p>
<details class="sd-sphinx-override sd-dropdown sd-card sd-mb-3">
<summary class="sd-summary-title sd-card-header sd-bg-info sd-bg-text-info">
Bowtie2 - extracts from <a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a>:<div class="sd-summary-down docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="sd-summary-up docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="sd-summary-content sd-card-body docutils">
<p class="sd-card-text">Here you can see extracts from the Bowtie 2 manual, available at <a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a>.</p>
<p class="sd-card-text">Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences. It is particularly good at aligning reads of about 50 up to 100s or 1,000s of characters to relatively long (e.g. mammalian) genomes. Bowtie 2 indexes the genome with an FM Index (based on the Burrows-Wheeler Transform or BWT) to keep its memory footprint small: for the human genome, its memory footprint is typically around 3.2 gigabytes of RAM. Bowtie 2 supports gapped, local, and paired-end alignment modes.</p>
<p class="sd-card-text">Bowtie 2 outputs alignments in SAM format, enabling interoperation with a large number of other tools (e.g. SAMtools, GATK) that use SAM. Bowtie 2 is distributed under the GPLv3 license, and it runs on the command line under Windows, Mac OS X and Linux and BSD.</p>
<p class="sd-card-text">Bowtie 2 is often the first step in pipelines for comparative genomics, including for variation calling, ChIP-seq, RNA-seq, BS-seq. Bowtie 2 and Bowtie (also called “Bowtie 1” here) are also tightly integrated into many other tools, some of which are listed here.</p>
<p class="sd-card-text"><strong>THE bowtie2 ALIGNER</strong></p>
<p class="sd-card-text"><code class="docutils literal notranslate"><span class="pre">bowtie2</span></code> takes a Bowtie 2 index and a set of sequencing read files and outputs a set of alignments in SAM format.</p>
<p class="sd-card-text">“Alignment” is the process by which we discover how and where the read sequences are similar to the reference sequence. An “alignment” is a result from this process, specifically: an alignment is a way of “lining up” some or all of the characters in the read with some characters from the reference in a way that reveals how they’re similar. For example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Read</span><span class="p">:</span>       <span class="n">GACTGGGCGATCTCGACTTCG</span>
            <span class="o">|||||</span>  <span class="o">||||||||||</span> <span class="o">|||</span>
<span class="n">Reference</span><span class="p">:</span>  <span class="n">GACTG</span><span class="o">--</span><span class="n">CGATCTCGACATCG</span>
</pre></div>
</div>
<p class="sd-card-text">Where dash symbols represent gaps and vertical bars show where aligned characters match.</p>
<p class="sd-card-text">End-to-end alignment versus local alignment</p>
<p class="sd-card-text">By default, Bowtie 2 performs end-to-end read alignment. That is, it searches for alignments involving all of the read characters. This is also called an “untrimmed” or “unclipped” alignment.</p>
<p class="sd-card-text">When the <code class="docutils literal notranslate"><span class="pre">--local</span></code> option is specified, Bowtie 2 performs local read alignment. In this mode, Bowtie 2 might “trim” or “clip” some read characters from one or both ends of the alignment if doing so maximizes the alignment score.</p>
<p class="sd-card-text"><strong>Scores: higher = more similar</strong></p>
<p class="sd-card-text">An alignment score quantifies how similar the read sequence is to the reference sequence aligned to. The higher the score, the more similar they are. A score is calculated by subtracting penalties for each difference (mismatch, gap, etc.) and, in local alignment mode, adding bonuses for each match.</p>
<p class="sd-card-text"><code class="docutils literal notranslate"><span class="pre">--score-min</span></code> Sets a function governing the minimum alignment score needed for an alignment to be considered “valid” (i.e. good enough to report). This is a function of read length. For instance, specifying L,0,-0.6 sets the minimum-score function f to f(x) = 0 + -0.6 * x, where x is the read length. See also: setting function options. The default in <code class="docutils literal notranslate"><span class="pre">--end-to-end</span></code> mode is L,-0.6,-0.6 and the default in <code class="docutils literal notranslate"><span class="pre">--local</span></code> mode is G,20,8.</p>
<p class="sd-card-text">from: <a class="reference external" href="https://github.com/BenLangmead/bowtie2/blob/master/MANUAL">https://github.com/BenLangmead/bowtie2/blob/master/MANUAL</a></p>
</div>
</details><details class="sd-sphinx-override sd-dropdown sd-card sd-mb-3">
<summary class="sd-summary-title sd-card-header sd-bg-info sd-bg-text-info">
Samtools: extracts from <a class="reference external" href="http://www.htslib.org">http://www.htslib.org</a><div class="sd-summary-down docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="sd-summary-up docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="sd-summary-content sd-card-body docutils">
<p class="sd-card-text">Here you can see extracts from Samtools manual, available here <a class="reference external" href="http://www.htslib.org">http://www.htslib.org</a>.</p>
<p class="sd-card-text">Samtools is a suite of programs for interacting with high-throughput sequencing data. It consists of three separate repositories:</p>
<p class="sd-card-text">Samtools: Reading/writing/editing/indexing/viewing SAM/BAM/CRAM format</p>
<p class="sd-card-text">BCFtools: Reading/writing BCF2/VCF/gVCF files and calling/filtering/summarising SNP and short indel sequence variants</p>
<p class="sd-card-text">HTSlib: A C library for reading/writing high-throughput sequencing data</p>
<p class="sd-card-text">Samtools and BCFtools both use HTSlib internally, but these source packages contain their own copies of htslib so they can be built independently.</p>
<p class="sd-card-text">Samtools is a set of utilities that manipulate alignments in the SAM (Sequence Alignment/Map), BAM, and CRAM formats. It converts between the formats, does sorting, merging and indexing, and can retrieve reads in any regions swiftly.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>samtools view -bS <span class="nv">$sam</span> <span class="p">|</span> samtools sort -o <span class="nv">$_sorted</span>.bam <span class="nv">$sam</span>
</pre></div>
</div>
<p class="sd-card-text"><code class="docutils literal notranslate"><span class="pre">view</span></code>: With no options or regions specified, prints all alignments in the specified input alignment file (in SAM, BAM, or CRAM format) to standard output in SAM format (with no header). <code class="docutils literal notranslate"><span class="pre">-b</span></code> Output in the BAM format. <code class="docutils literal notranslate"><span class="pre">-S</span></code> Ignored for compatibility with previous samtools versions. Previously this option was required if input was in SAM format, but now the correct format is automatically detected by examining the first few characters of input.</p>
<p class="sd-card-text"><code class="docutils literal notranslate"><span class="pre">sort</span></code>: Sort alignments by leftmost coordinates, or by read name when <code class="docutils literal notranslate"><span class="pre">-n</span></code> is used. An appropriate &#64;HD-SO sort order header tag will be added or an existing one updated if necessary. <code class="docutils literal notranslate"><span class="pre">-o</span></code> FILE Write the final sorted output to FILE, rather than to standard output.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>samtools index <span class="nv">$index</span>
</pre></div>
</div>
<p class="sd-card-text"><code class="docutils literal notranslate"><span class="pre">index</span></code>: Index a coordinate-sorted BGZIP-compressed SAM, BAM or CRAM file for fast random access. Note for SAM this only works if the file has been BGZF compressed first.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>samtools mpileup -E -uf Ref.fna <span class="nv">$index</span> &gt; <span class="nv">$pileup</span>
</pre></div>
</div>
<p class="sd-card-text">BCFtools is a set of utilities that manipulate variant calls in the Variant Call Format (VCF) and its binary counterpart BCF. All commands work transparently with both VCFs and BCFs, both uncompressed and BGZF-compressed.</p>
<p class="sd-card-text"><code class="docutils literal notranslate"><span class="pre">mpileup</span></code>: Generate VCF or BCF containing genotype likelihoods for one or multiple alignment (BAM or CRAM) files. This is based on the original samtools mpileup command (with the -v or -g options) producing genotype likelihoods in VCF or BCF format, but not the textual pileup output. <strong>The mpileup command was transferred to bcftools in order to avoid errors resulting from use of incompatible versions of samtools and bcftools when using in the mpileup+bcftools call pipeline.</strong></p>
<p class="sd-card-text">Generate VCF, BCF or pileup for one or multiple BAM files. Alignment records are grouped by sample (SM) identifiers in &#64;RG header lines. If sample identifiers are absent, each input file is regarded as one sample. <code class="docutils literal notranslate"><span class="pre">-E</span></code>, –redo-BAQ Recalculate BAQ on the fly, ignore existing BQ tags <code class="docutils literal notranslate"><span class="pre">-uF</span></code> -f, –fasta-ref FILE The faidx-indexed reference file in the FASTA format. The file can be optionally compressed by bgzip.</p>
<p class="sd-card-text">Samtools mpileup can still produce VCF and BCF output (with -g or -u), but this feature is deprecated and will be removed in a future release. Please use bcftools mpileup for this instead.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>bcftools call -c <span class="nv">$pileup</span> &gt; <span class="nv">$vcf</span><span class="sb">`</span>
</pre></div>
</div>
<p class="sd-card-text">This command replaces the former bcftools view caller. Some of the original functionality has been temporarily lost in the process of transition under htslib, but will be added back on popular demand. The original calling model can be invoked with the -c option.</p>
</div>
</details><p>First, we need to index the bait set for Bowtie2:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>bowtie2-build Ref.fna Inga_unique_baits
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Here the name <em>Inga_unique_baits</em> is a choice of the user. It will be the name of the indexed baits that Bowtie2 will use for the analysis. You can choose any name of your preference but remember to change accordingly in the <code class="docutils literal notranslate"><span class="pre">bam_me.sh</span></code> script, more specifically in the line: <code class="docutils literal notranslate"><span class="pre">bowtie2</span> <span class="pre">--local</span> <span class="pre">--score-min</span> <span class="pre">$score</span> <span class="pre">-x</span> <span class="pre">./Inga_unique_baits</span> <span class="pre">-1</span> <span class="pre">$fwd_p</span>&#160; <span class="pre">-2</span> <span class="pre">$rev_p</span>&#160; <span class="pre">-U</span> <span class="pre">$un_p</span>&#160; <span class="pre">-S</span> <span class="pre">$sam</span> <span class="pre">2&gt;$bowtie</span></code>. <code class="docutils literal notranslate"><span class="pre">Ref.fna</span></code> is the fasta file containing the bait sequence, we will still need it in a later <code class="docutils literal notranslate"><span class="pre">samtools</span></code> step.</p>
</div>
<p>Now we are ready to run the <code class="docutils literal notranslate"><span class="pre">bam_me.sh</span></code> script in a loop:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="k">while</span> <span class="nb">read</span> f<span class="p">;</span> <span class="k">do</span> ./bam_me.sh <span class="s2">&quot;</span><span class="nv">$f</span><span class="s2">&quot;</span> <span class="p">;</span> <span class="k">done</span> &lt; acc
</pre></div>
</div>
<p>In our folder we now have:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>FG113                                 FG35_1.fastq.gz                       FGIntype_2.fastq.gz                   KGD465_1.fastq.gz                     fastqctrimfile
FG113.empties                         FG35_1_fastqc.html                    FGIntype_2_fastqc.html                KGD465_1_fastqc.html                  more_tidying.sh
FG113.vcf                             FG35_1_fastqc.zip                     FGIntype_2_fastqc.zip                 KGD465_1_fastqc.zip                   renaming.sh
FG113_1.fastq.gz                      FG35_2.fastq.gz                       FGIntype_bowtie_output                KGD465_2.fastq.gz                     zygia917
FG113_1_fastqc.html                   FG35_2_fastqc.html                    FGIntype_forward_paired.fq.gz         KGD465_2_fastqc.html                  zygia917.empties
FG113_1_fastqc.zip                    FG35_2_fastqc.zip                     FGIntype_forward_paired_fastqc.html   KGD465_2_fastqc.zip                   zygia917.vcf
FG113_2.fastq.gz                      FG35_bowtie_output                    FGIntype_forward_paired_fastqc.zip    KGD465_bowtie_output                  zygia917_1.fastq.gz
FG113_2_fastqc.html                   FG35_forward_paired.fq.gz             FGIntype_forward_unpaired.fq.gz       KGD465_forward_paired.fq.gz           zygia917_1_fastqc.html
FG113_2_fastqc.zip                    FG35_forward_paired_fastqc.html       FGIntype_forward_unpaired_fastqc.html KGD465_forward_paired_fastqc.html     zygia917_1_fastqc.zip
FG113_bowtie_output                   FG35_forward_paired_fastqc.zip        FGIntype_forward_unpaired_fastqc.zip  KGD465_forward_paired_fastqc.zip      zygia917_2.fastq.gz
FG113_forward_paired.fq.gz            FG35_forward_unpaired.fq.gz           FGIntype_reverse_paired.fq.gz         KGD465_forward_unpaired.fq.gz         zygia917_2_fastqc.html
FG113_forward_paired_fastqc.html      FG35_forward_unpaired_fastqc.html     FGIntype_reverse_paired_fastqc.html   KGD465_forward_unpaired_fastqc.html   zygia917_2_fastqc.zip
FG113_forward_paired_fastqc.zip       FG35_forward_unpaired_fastqc.zip      FGIntype_reverse_paired_fastqc.zip    KGD465_forward_unpaired_fastqc.zip    zygia917_bowtie_output
FG113_forward_unpaired.fq.gz          FG35_reverse_paired.fq.gz             FGIntype_reverse_unpaired.fq.gz       KGD465_reverse_paired.fq.gz           zygia917_forward_paired.fq.gz
FG113_forward_unpaired_fastqc.html    FG35_reverse_paired_fastqc.html       FGIntype_reverse_unpaired_fastqc.html KGD465_reverse_paired_fastqc.html     zygia917_forward_paired_fastqc.html
FG113_forward_unpaired_fastqc.zip     FG35_reverse_paired_fastqc.zip        FGIntype_reverse_unpaired_fastqc.zip  KGD465_reverse_paired_fastqc.zip      zygia917_forward_paired_fastqc.zip
FG113_reverse_paired.fq.gz            FG35_reverse_unpaired.fq.gz           FGIntype_sorted.bam                   KGD465_reverse_unpaired.fq.gz         zygia917_forward_unpaired.fq.gz
FG113_reverse_paired_fastqc.html      FG35_reverse_unpaired_fastqc.html     FGIntype_sorted.bam.bai               KGD465_reverse_unpaired_fastqc.html   zygia917_forward_unpaired_fastqc.html
FG113_reverse_paired_fastqc.zip       FG35_reverse_unpaired_fastqc.zip      FGIntype_trimmed_1.fastq              KGD465_reverse_unpaired_fastqc.zip    zygia917_forward_unpaired_fastqc.zip
FG113_reverse_unpaired.fq.gz          FG35_sorted.bam                       FGIntype_trimmed_1.fastq.gz           KGD465_sorted.bam                     zygia917_reverse_paired.fq.gz
FG113_reverse_unpaired_fastqc.html    FG35_sorted.bam.bai                   FGIntype_trimmed_1u.fastq             KGD465_sorted.bam.bai                 zygia917_reverse_paired_fastqc.html
FG113_reverse_unpaired_fastqc.zip     FG35_trimmed_1.fastq                  FGIntype_trimmed_2.fastq              KGD465_trimmed_1.fastq                zygia917_reverse_paired_fastqc.zip
FG113_sorted.bam                      FG35_trimmed_1.fastq.gz               FGIntype_trimmed_2.fastq.gz           KGD465_trimmed_1.fastq.gz             zygia917_reverse_unpaired.fq.gz
FG113_sorted.bam.bai                  FG35_trimmed_1u.fastq                 FGIntype_trimmed_2u.fastq             KGD465_trimmed_1u.fastq               zygia917_reverse_unpaired_fastqc.html
FG113_trimmed_1.fastq                 FG35_trimmed_2.fastq                  Inga_unique_baits.1.bt2               KGD465_trimmed_2.fastq                zygia917_reverse_unpaired_fastqc.zip
FG113_trimmed_1.fastq.gz              FG35_trimmed_2.fastq.gz               Inga_unique_baits.2.bt2               KGD465_trimmed_2.fastq.gz             zygia917_sorted.bam
FG113_trimmed_1u.fastq                FG35_trimmed_2u.fastq                 Inga_unique_baits.3.bt2               KGD465_trimmed_2u.fastq               zygia917_sorted.bam.bai
FG113_trimmed_2.fastq                 FGIntype                              Inga_unique_baits.4.bt2               Ref.fna                               zygia917_trimmed_1.fastq
FG113_trimmed_2.fastq.gz              FGIntype.empties                      Inga_unique_baits.rev.1.bt2           Ref.fna.fai                           zygia917_trimmed_1.fastq.gz
FG113_trimmed_2u.fastq                FGIntype.vcf                          Inga_unique_baits.rev.2.bt2           acc                                   zygia917_trimmed_1u.fastq
FG35                                  FGIntype_1.fastq.gz                   KGD465                                bam_me.sh                             zygia917_trimmed_2.fastq
FG35.empties                          FGIntype_1_fastqc.html                KGD465.empties                        cut_out                               zygia917_trimmed_2.fastq.gz
FG35.vcf                              FGIntype_1_fastqc.zip                 KGD465.vcf                            fastqcfiles                           zygia917_trimmed_2u.fastq
</pre></div>
</div>
<p>The file <code class="docutils literal notranslate"><span class="pre">{$acc}_bowtie_output</span></code> shows the aligment summary for the bowtie2 step. We can take a look inside those file with:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>awk <span class="s1">&#39;FNR==1{print &quot;::::\n&quot;FILENAME&quot;\n::::&quot;}1&#39;</span> *_bowtie_output
</pre></div>
</div>
<p>You can also visually explore the bam files using the software <a class="reference external" href="https://ics.hutton.ac.uk/tablet/">Tablet</a>. You will need the <code class="docutils literal notranslate"><span class="pre">.bam</span></code> and the index <code class="docutils literal notranslate"><span class="pre">.bai</span></code> files, which were generated with the <code class="docutils literal notranslate"><span class="pre">bam_me.sh</span></code> script. Below is an example of what it looks like. It is a good idea to explore the <code class="docutils literal notranslate"><span class="pre">.bam</span></code> files to check how the reads mapped back to the baits.</p>
<img alt="This is an example of a ``.bam`` file viewed with Tablet." src="_images/bam_file_example.png" />
<p>One of the outputs this script produces is the vcf file. VCF is the standard file format for storing variation data.</p>
<details class="sd-sphinx-override sd-dropdown sd-card sd-mb-3">
<summary class="sd-summary-title sd-card-header sd-bg-info sd-bg-text-info">
What are VCF files<div class="sd-summary-down docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="sd-summary-up docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="sd-summary-content sd-card-body docutils">
<p class="sd-card-text">The Variant Call Format (VCF) specifies the format of a text file used in bioinformatics for storing gene sequence variations. The format has been developed with the advent of large-scale genotyping and DNA sequencing projects, such as the 1000 Genomes Project. Existing formats for genetic data such as General feature format (GFF) stored all of the genetic data, much of which is redundant because it will be shared across the genomes. By using the variant call format only the variations need to be stored along with a reference genome. More info:
<a class="reference external" href="http://www.internationalgenome.org/wiki/Analysis/vcf4.0">http://www.internationalgenome.org/wiki/Analysis/vcf4.0</a> and
<a class="reference external" href="http://augustogarcia.me/statgen-esalq/Hapmap-and-VCF-formats-and-its-integration-with-onemap/">http://augustogarcia.me/statgen-esalq/Hapmap-and-VCF-formats-and-its-integration-with-onemap/</a></p>
</div>
</details><p>We will use the script <code class="docutils literal notranslate"><span class="pre">clean_vcf.sh</span></code> to edit the vcf files to remove indels and calls with quality less than 36 and then output the consensus fasta. Note that we will need the perl script <code class="docutils literal notranslate"><span class="pre">vcfutils_dasta.pl</span></code> in the folder we are working on.</p>
<details class="sd-sphinx-override sd-dropdown sd-card sd-mb-3">
<summary class="sd-summary-title sd-card-header sd-bg-info sd-bg-text-info">
clean_vcf.sh<div class="sd-summary-down docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="sd-summary-up docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="sd-summary-content sd-card-body docutils">
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#! /bin/bash -x</span>
<span class="c1"># to remove tricky stuff from the vcf files and make the fasta</span>
<span class="c1"># Input on comnad line is the accession stem</span>
<span class="c1">#Catherine Kidner 11 Nov 2014</span>


<span class="nb">echo</span> <span class="s2">&quot;Hello world&quot;</span>

<span class="nv">acc</span><span class="o">=</span><span class="nv">$1</span>

<span class="nb">echo</span> <span class="s2">&quot;You&#39;re working on accession </span><span class="nv">$1</span><span class="s2">&quot;</span>


<span class="nv">vcf</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>.vcf
<span class="nv">clean</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>_clean.vcf
<span class="nv">fasta</span><span class="o">=</span><span class="si">${</span><span class="nv">acc</span><span class="si">}</span>.fasta

grep -v <span class="s2">&quot;INDEL&quot;</span> <span class="nv">$vcf</span> <span class="p">|</span> awk <span class="s1">&#39;{if ($6 &gt;= 36) print $0}&#39;</span> &gt; <span class="nv">$clean</span>

perl vcfutils_fasta.pl vcf2fq <span class="nv">$clean</span> &gt; <span class="nv">$fasta</span>

rm <span class="nv">$clean</span>

<span class="nb">exit</span> <span class="m">0</span>
</pre></div>
</div>
</div>
</details><p>We now have one fasta file per accession, and each file contains the recovered sequence for each of the 264 <em>loci</em>.</p>
<p>We will convert from multifastas of loci per accession to multifasta of accession per locus using the python script <code class="docutils literal notranslate"><span class="pre">switch_multifastas.py</span></code>. This script needs a folder called By_locus, a list of loci called <code class="docutils literal notranslate"><span class="pre">locus_list</span></code> and a list of files called <code class="docutils literal notranslate"><span class="pre">fasta_files</span></code>.</p>
<p>Let’s create the folder and lists:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir By_locus
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cat Ref.fna <span class="p">|</span> grep <span class="s2">&quot;&gt;&quot;</span> <span class="p">|</span> sed <span class="s1">&#39;s/\&gt;//g&#39;</span> &gt; locus_list
</pre></div>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ls *.fasta &gt; fasta_files
</pre></div>
</div>
<p>And then run:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>python /path/to/your/Targeted_enrichment/Switch_multifastas.py
</pre></div>
</div>
<details class="sd-sphinx-override sd-dropdown sd-card sd-mb-3">
<summary class="sd-summary-title sd-card-header sd-bg-info sd-bg-text-info">
Switch_multifastas.py<div class="sd-summary-down docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-down" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M5.22 8.72a.75.75 0 000 1.06l6.25 6.25a.75.75 0 001.06 0l6.25-6.25a.75.75 0 00-1.06-1.06L12 14.44 6.28 8.72a.75.75 0 00-1.06 0z"></path></svg></div>
<div class="sd-summary-up docutils">
<svg version="1.1" width="1.5em" height="1.5em" class="sd-octicon sd-octicon-chevron-up" viewBox="0 0 24 24" aria-hidden="true"><path fill-rule="evenodd" d="M18.78 15.28a.75.75 0 000-1.06l-6.25-6.25a.75.75 0 00-1.06 0l-6.25 6.25a.75.75 0 101.06 1.06L12 9.56l5.72 5.72a.75.75 0 001.06 0z"></path></svg></div>
</summary><div class="sd-summary-content sd-card-body docutils">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="sd">&#39;&#39;&#39;take a set of multifastas and re-order as by filename a set of multifastas</span>
<span class="sd">For dealing with hyb_baits bowtie to fasta outputs</span>
<span class="sd">needs alist f the fasta file names and a list of the loci (as in the fasta_files)</span>
<span class="sd">set at the default - can change to querry for list names</span>
<span class="sd">needs an output folder called By_locus&#39;&#39;&#39;</span>


<span class="k">def</span> <span class="nf">fasta_dict</span><span class="p">(</span><span class="n">fastafile</span><span class="p">):</span>
      <span class="n">file</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="n">fastafile</span><span class="p">)</span>
      <span class="n">name2seq</span> <span class="o">=</span> <span class="p">{}</span>

      <span class="k">for</span> <span class="n">line</span> <span class="ow">in</span> <span class="n">file</span><span class="p">:</span>
            <span class="k">if</span> <span class="n">line</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;&gt;&quot;</span><span class="p">):</span>
                  <span class="n">seq</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>
                  <span class="n">split_line</span> <span class="o">=</span> <span class="n">line</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">)</span>
                  <span class="n">name</span> <span class="o">=</span> <span class="n">split_line</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
                  <span class="n">name</span> <span class="o">=</span> <span class="n">name</span><span class="o">.</span><span class="n">rstrip</span><span class="p">()</span>
                  <span class="n">name</span> <span class="o">=</span> <span class="n">name</span><span class="o">.</span><span class="n">lstrip</span><span class="p">(</span><span class="s1">&#39;&gt;&#39;</span><span class="p">)</span>
            <span class="k">else</span><span class="p">:</span>
                  <span class="n">seq</span> <span class="o">=</span> <span class="n">seq</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">line</span><span class="p">)</span>
                  <span class="n">seq</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">rstrip</span><span class="p">()</span>

            <span class="n">name2seq</span><span class="p">[</span><span class="n">name</span><span class="p">]</span> <span class="o">=</span> <span class="n">seq</span>

      <span class="k">return</span> <span class="n">name2seq</span>

<span class="nb">print</span> <span class="p">(</span><span class="s2">&quot;Hello world&quot;</span><span class="p">)</span>

<span class="n">fastafile_list</span> <span class="o">=</span> <span class="s2">&quot;fasta_files&quot;</span>
<span class="n">locus_list_file</span> <span class="o">=</span> <span class="s2">&quot;locus_list&quot;</span>

<span class="c1">#fastafile_list = input(&quot;Which list of fasta files (full file name please)?\n&quot;)</span>
<span class="c1">#locus_list_file = input(&quot;Which file for names of loci?\n&quot;)</span>

<span class="c1">#open files for each locus (for reading and writting) ready to feed in the fastas</span>

<span class="k">if</span> <span class="n">locus_list_file</span> <span class="o">!=</span> <span class="kc">None</span><span class="p">:</span>
      <span class="n">locus_list</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="n">locus_list_file</span><span class="p">)</span>
      <span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">locus_list</span><span class="p">:</span>
            <span class="n">name</span> <span class="o">=</span> <span class="n">name</span><span class="o">.</span><span class="n">rstrip</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
            <span class="n">out_file_name</span> <span class="o">=</span> <span class="n">name</span> <span class="o">+</span> <span class="s2">&quot;.fasta&quot;</span>

            <span class="n">outfile</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="n">out_file_name</span><span class="p">,</span> <span class="s2">&quot;w&quot;</span><span class="p">)</span>
            <span class="n">outfile</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>

<span class="c1"># go through each fasta file pulling out the match to each locus and putting into the file</span>

<span class="k">if</span> <span class="n">fastafile_list</span> <span class="o">!=</span> <span class="kc">None</span><span class="p">:</span>
      <span class="n">fastafiles</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="n">fastafile_list</span><span class="p">)</span>
      <span class="k">for</span> <span class="n">file_name</span> <span class="ow">in</span> <span class="n">fastafiles</span><span class="p">:</span>
            <span class="n">file_name</span> <span class="o">=</span> <span class="n">file_name</span><span class="o">.</span><span class="n">rstrip</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
            <span class="n">accession</span> <span class="o">=</span> <span class="n">file_name</span><span class="o">.</span><span class="n">rstrip</span><span class="p">(</span><span class="s2">&quot;.fna&quot;</span><span class="p">)</span>

            <span class="c1">#Make a dict of that fasta file</span>

            <span class="n">ifasta_dict</span> <span class="o">=</span> <span class="n">fasta_dict</span><span class="p">(</span><span class="n">file_name</span><span class="p">)</span>

            <span class="n">keep_lines_processed</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="n">keep_seq_found</span> <span class="o">=</span> <span class="mi">0</span>
            <span class="n">missing_list</span> <span class="o">=</span> <span class="p">[]</span>

            <span class="k">if</span> <span class="n">locus_list_file</span> <span class="o">!=</span> <span class="kc">None</span><span class="p">:</span>
                  <span class="n">locus_list</span> <span class="o">=</span> <span class="nb">open</span><span class="p">(</span><span class="n">locus_list_file</span><span class="p">)</span>
                  <span class="c1">#for a locus pull out the hit and append to the correct file</span>
                  <span class="k">for</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">locus_list</span><span class="p">:</span>
                        <span class="n">name</span> <span class="o">=</span> <span class="n">name</span><span class="o">.</span><span class="n">rstrip</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="p">)</span>
                        <span class="n">found_fasta</span> <span class="o">=</span> <span class="s2">&quot;&gt;&quot;</span> <span class="o">+</span>  <span class="n">accession</span> <span class="o">+</span> <span class="s2">&quot;_&quot;</span> <span class="o">+</span> <span class="n">name</span> <span class="o">+</span> <span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span>
                        <span class="n">keep_lines_processed</span> <span class="o">+=</span> <span class="mi">1</span>


                        <span class="n">fasta_seq</span> <span class="o">=</span> <span class="n">ifasta_dict</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">name</span><span class="p">,</span> <span class="s2">&quot;NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN&quot;</span><span class="p">)</span>
                        <span class="k">if</span> <span class="n">name</span> <span class="ow">in</span> <span class="n">ifasta_dict</span><span class="o">.</span><span class="n">keys</span><span class="p">():</span>
                              <span class="n">keep_seq_found</span> <span class="o">+=</span><span class="mi">1</span>
                        <span class="k">else</span><span class="p">:</span>
                              <span class="n">missing_list</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">name</span><span class="p">)</span>

                        <span class="n">found_fasta</span> <span class="o">=</span> <span class="n">found_fasta</span> <span class="o">+</span> <span class="n">fasta_seq</span> <span class="o">+</span> <span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span>
                        <span class="n">locus_file_to_write</span> <span class="o">=</span> <span class="s2">&quot;By_locus/&quot;</span> <span class="o">+</span> <span class="n">name</span> <span class="o">+</span> <span class="s2">&quot;.fasta&quot;</span>
                        <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="n">locus_file_to_write</span><span class="p">,</span> <span class="s2">&quot;a&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">myfile</span><span class="p">:</span>
                              <span class="n">myfile</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">found_fasta</span><span class="p">)</span>



            <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Keep lines processed for &quot;</span> <span class="o">+</span> <span class="n">file_name</span> <span class="o">+</span> <span class="s2">&quot;  = &quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">keep_lines_processed</span><span class="p">))</span>
            <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Keep sequences found for &quot;</span> <span class="o">+</span> <span class="n">file_name</span> <span class="o">+</span> <span class="s2">&quot; = &quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="n">keep_seq_found</span><span class="p">))</span>
            <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Missing for &quot;</span> <span class="o">+</span> <span class="n">file_name</span> <span class="o">+</span> <span class="s2">&quot; = &quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\n</span><span class="s2">&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">missing_list</span><span class="p">)))</span>
</pre></div>
</div>
</div>
</details><p>We now have a folder callled <code class="docutils literal notranslate"><span class="pre">By_locus</span></code> with 264 fasta files ready to be aligned with <a class="reference external" href="https://mafft.cbrc.jp/alignment/software/">mafft</a>.</p>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="cutadapt.html" class="btn btn-neutral float-left" title="Cutadapt" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="aligning.html" class="btn btn-neutral float-right" title="Alignment" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021 | Bioinformatics | Royal Botanic Garden Edinburgh.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>